google_analytics/notebooks/load.ipynb

```diff
@@ -34,24 +34,189 @@
+    "num_purchases = len(single_visits[single_visits[\"transactionRevenue\"] > 0][\"transactionRevenue\"])\n",
+    "print(\"Out of {} single visits, {} ended up in a purchase\".format(len(single_visit_ids), num_purchases))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 68,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "We will therefore choose to ignore 262759 users\n"
+     ]
+    }
+   ],
+   "source": [
+    "to_ignore = set(single_visit_ids) - set(single_visits[single_visits[\"transactionRevenue\"] > 0][\"fullVisitorId\"])\n",
+    "print(\"We will therefore choose to ignore {} users\".format(len(to_ignore)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Is this assumption justified?\n",
+    "\n",
+    "How reliable is this assumption? Are these people that we choose to ignore indeed buying nothing? Let's find out.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 72,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "There are 2290 people appearing both in the training and the test periods\n",
+      "We will ignore 2037 out of those 2290 people\n"
+     ]
+    }
+   ],
+   "source": [
+    "# We will only focus on people who exist in both datasets\n",
+    "common_ids = set(y_train[\"fullVisitorId\"]).intersection(x_train[\"fullVisitorId\"])\n",
+    "print(\"There are {} people appearing both in the training and the test periods\".format(len(common_ids)))\n",
     "\n",
-    "# preprocess_and_save(\"../data\", nrows_train=100000, nrows_test=10000)\n",
-    "train, test = process(train, test)"
+    "y_train = y_train[y_train[\"fullVisitorId\"].isin(common_ids)]\n",
+    "ignored = y_train[y_train[\"fullVisitorId\"].isin(to_ignore)]\n",
+    "print(\"We will ignore {} out of those {} people\".format(len(ignored), len(common_ids)))"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 88,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "We made 196 mistakes (9.62% of cases) leading to loss = 5.4433800738203315\n"
+     ]
+    }
+   ],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "predicted_target = [0] * len(common_ids)\n",
+    "\n",
+    "y_train_grouped = y_train[[\"fullVisitorId\", \"transactionRevenue\"]].groupby(\"fullVisitorId\", as_index=False).sum()\n",
+    "actual_target = y_train_grouped[\"transactionRevenue\"].apply(lambda revenue: np.log(revenue + 1))\n",
+    "    \n",
+    "loss = np.sqrt(np.mean((predicted_target - actual_target)**2))\n",
+    "num_mistakes = sum(actual_target > 0)\n",
+    "\n",
+    "print(\"We made {} mistakes ({:.2f}% of cases) leading to loss = {}\".format(num_mistakes, 100*num_mistakes/len(ignored) ,loss))"
```

Comment:
Very nice that you translated the misclassification into the actual error on the leaderboard!